app/code/AKisilenko/ModuleLesson6/Model/AskQuestionRepository.php

```diff
@@ -0,0 +1,188 @@
+     * @var AskQuestionInterfaceFactory
+     */
+    protected $dataAskQuestionFactory;
+
+    /**
+     * @var AskQuestionSearchResultsInterfaceFactory
+     */
+    protected $searchResultsFactory;
+
+    /**
+     * @var DataObjectHelper
+     */
+    protected $dataObjectHelper;
+
+    /**
+     * @var DataObjectProcessor
+     */
+    protected $dataObjectProcessor;
+
+    /**
+     * AskQuestionRepository constructor.
+     * @param ResourceAskQuestion $resource
+     * @param AskQuestionFactory $askQuestionFactory
+     * @param AskQuestionCollectionFactory $askQuestionCollectionFactory
+     * @param AskQuestionInterfaceFactory $dataAskQuestionFactory
+     * @param AskQuestionSearchResultsInterfaceFactory $searchResultsFactory
+     * @param DataObjectHelper $dataObjectHelper
+     * @param DataObjectProcessor $dataObjectProcessor
+     */
+    public function __construct(
+        ResourceAskQuestion $resource,
+        AskQuestionFactory $askQuestionFactory,
+        AskQuestionCollectionFactory $askQuestionCollectionFactory,
+        AskQuestionInterfaceFactory $dataAskQuestionFactory,
+        AskQuestionSearchResultsInterfaceFactory $searchResultsFactory,
+        DataObjectHelper $dataObjectHelper,
+        DataObjectProcessor $dataObjectProcessor
+    ) {
+        $this->resource = $resource;
+        $this->askQuestionFactory = $askQuestionFactory;
+        $this->askQuestionCollectionFactory = $askQuestionCollectionFactory;
+        $this->searchResultsFactory = $searchResultsFactory;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->dataAskQuestionFactory = $dataAskQuestionFactory;
+        $this->dataObjectProcessor = $dataObjectProcessor;
+    }
+
+    /**
+     * @param AskQuestionInterface $askQuestion
+     * @return AskQuestionInterface|mixed
+     * @throws CouldNotSaveException
+     */
+    public function save(AskQuestionInterface $askQuestion)
+    {
+        try {
+            $this->resource->save($askQuestion);
+        } catch (Exception $exception) {
+            throw new CouldNotSaveException(__($exception->getMessage()));
+        }
+        return $askQuestion;
+    }
+
+    /**
+     * @param $askQuestionId
+     * @return mixed
+     * @throws NoSuchEntityException
+     */
+    public function getById($askQuestionId)
+    {
+        $askQuestion = $this->askQuestionFactory->create();
+        $this->resource->load($askQuestion, $askQuestionId);
+        if (!$askQuestion->getId()) {
+            throw new NoSuchEntityException(__('Question with id "%1" does not exist.', $askQuestionId));
+        }
+        return $askQuestion;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $criteria
+     * @return mixed
+     */
+    public function getList(SearchCriteriaInterface $criteria)
+    {
+        $searchResults = $this->searchResultsFactory->create();
+        $searchResults->setSearchCriteria($criteria);
+        $collection = $this->AskQuestionCollectionFactory->create();
```

Comment:
Typo here: 
 $collection = $this->askQuestionCollectionFactory->create();
(field name begins with lowercase a: askQ.. not AskQ